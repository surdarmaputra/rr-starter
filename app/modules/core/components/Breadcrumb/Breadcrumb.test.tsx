import { render, screen, within } from '@testing-library/react';
import { Slash } from 'lucide-react';
import { describe, expect, test } from 'vitest';

import {
  Breadcrumb,
  BreadcrumbItem,
  BreadcrumbLink,
  BreadcrumbList,
  BreadcrumbSeparator,
} from './Breadcrumb';

describe('Breadcrumb', () => {
  function renderComponent() {
    return render(
      <Breadcrumb data-testid="Breadcrumb-testid">
        <BreadcrumbList>
          <BreadcrumbItem>
            <BreadcrumbLink href="/">Home</BreadcrumbLink>
          </BreadcrumbItem>
          <BreadcrumbSeparator>
            <Slash />
          </BreadcrumbSeparator>
          <BreadcrumbItem>
            <BreadcrumbLink href="/components">Components</BreadcrumbLink>
          </BreadcrumbItem>
        </BreadcrumbList>
      </Breadcrumb>,
    );
  }

  test('renders with data-testid', async () => {
    renderComponent();
    const breadcrumb = await screen.findByTestId('Breadcrumb-testid');
    within(breadcrumb).getByText(/Home/);
    within(breadcrumb).getByText(/Components/);
  });

  test('renders snapshot', async () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
