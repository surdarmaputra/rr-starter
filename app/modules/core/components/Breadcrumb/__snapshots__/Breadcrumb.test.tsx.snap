// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Breadcrumb > renders snapshot 1`] = `
<div>
  <nav
    aria-label="breadcrumb"
    data-testid="Breadcrumb-testid"
  >
    <ol
      class="flex flex-wrap items-center gap-1.5 break-words text-sm text-letter-caption"
    >
      <li
        class="inline-flex items-center gap-1.5"
      >
        <a
          class="transition-colors hover:text-letter-title"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        aria-hidden="true"
        class="[&>svg]:h-3.5 [&>svg]:w-3.5"
        role="presentation"
      >
        <svg
          class="lucide lucide-slash"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M22 2 2 22"
          />
        </svg>
      </li>
      <li
        class="inline-flex items-center gap-1.5"
      >
        <a
          class="transition-colors hover:text-letter-title"
          href="/components"
        >
          Components
        </a>
      </li>
    </ol>
  </nav>
</div>
`;
